src/functions.php

```diff
@@ -432,6 +432,18 @@ function is_valid_hostname($hostname)
     );
 }
 
+/**
+ * Checks if supplied parameter is a valid host label
+ *
+ * @param $label
+ * @return bool
+ */
+function is_valid_hostlabel($label)
+{
+    return preg_match("/^([a-z\d](-*[a-z\d])*)$/", $label)
+        && preg_match("/^.{1,63}$/", $label);
```

Comment:
Genuine question that I'm not sure how to find the answer to: how does preg_match with that specific input vary in computational time from just checking the length of the string and doing a less than or greater than?

Reply:
New regex includes the length validation within itself, so need to worry about this case.